test_scripts/Polices/build_options/039_ATF_P_TC_PTU_HMI_Level_Affected_Apps_FULL_LIMITED_HTTP.lua

```diff
@@ -111,97 +116,105 @@ function Test:TestStep_RegisterFirstApp()
+  :Do(function(_,data)
+      print("SDL -> MOB1: OnSystemRequest, requestType: " .. data.payload.requestType)
+      if(data.payload.requestType == "HTTP") then
+        if(onsystemreq_app1 == true) then self:FailTestCase("OnSystemRequest(HTTP) has already received for application 2") end
+        onsystemreq_app2 = true
       end
-      return false
-      end):Times(Between(1,2))
-    self.mobileSession:ExpectNotification("OnPermissionsChange")
-    -- Expect after updating HMI status will not change
-    self.mobileSession:ExpectNotification("OnHMIStatus"):Times(0)
-  end
-
-  function Test:TestStep_RegisterSecondApp()
-    self.mobileSession1 = mobile_session.MobileSession(self, self.mobileConnection)
-    self.mobileSession1:StartService(7)
-    :Do(function (_,_)
-        local correlationId = self.mobileSession1:SendRPC("RegisterAppInterface", config.application2.registerAppInterfaceParams)
-        EXPECT_HMINOTIFICATION("BasicCommunication.OnAppRegistered")
-        :Do(function(_,data)
-            HMIAppID = data.params.application.appID
-          end)
-        self.mobileSession1:ExpectResponse(correlationId, { success = true })
-        self.mobileSession1:ExpectNotification("OnPermissionsChange")
-      end)
-  end
-
-  function Test:TestStep_ActivateSecondAppInLimited()
-    commonSteps:ActivateAppInSpecificLevel(self,HMIAppID,"LIMITED")
-  end
-
-  function Test:TestStep_UpdatePolicyAfterAddSecondApp_ExpectOnHMIStatusNotCall()
-    self.mobileSession1:ExpectNotification("OnSystemRequest", {requestType = "HTTP"})
-    :Do(function()
-        local CorIdSystemRequest = self.mobileSession1:SendRPC("SystemRequest",
-          {
-            requestType = "HTTP",
-            fileName = "ptu2app.json",
-          },"files/ptu2app.json")
-        self.mobileSession1:ExpectResponse(CorIdSystemRequest, {success = true, resultCode = "SUCCESS"})
-      end)
-    EXPECT_HMINOTIFICATION("SDL.OnStatusUpdate")
-    :ValidIf(function(exp,data)
-        if
-        (exp.occurences == 1 or exp.occurences == 2) and
-        data.params.status == "UP_TO_DATE" then
-          return true
-        end
-        if
-        exp.occurences == 1 and
-        data.params.status == "UPDATING" then
-          return true
-        end
-        return false
-        end):Times(Between(1,2))
-      self.mobileSession1:ExpectNotification("OnPermissionsChange")
-      -- Expect after updating HMI status will not change
-      self.mobileSession1:ExpectNotification("OnHMIStatus"):Times(0)
-    end
-
-    --[[ Postconditions ]]
-    commonFunctions:newTestCasesGroup("Postconditions")
-
-    function Test.Postcondition_RemovePTUfiles()
-      os.remove(ptu_first_app_registered)
-      os.remove(ptu_second_app_registered)
-    end
-    function Test.Postcondition_Stop_SDL()
-      StopSDL()
-    end
-
-    return Test
+    end)
+
+  EXPECT_HMINOTIFICATION("SDL.OnStatusUpdate", {status = "UPDATE_NEEDED"}, {status = "UPDATING"}):Times(2)
```

Comment:
@istoimenova, Please add DelayedExp() in order to catch "OnSystemRequest". Because now "HTTP" request won't be caught since for Mob1 Between(1) and for Mob2 Between(0) are used